src/nix/nix_store.rs

```
@@ -0,0 +1,132 @@
+        }
+    }
+}
+
+impl fmt::Display for StorePath {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.as_path().display())
+    }
+}
+
+/// The `nix-store` command
+/// See documentation for [nix-store](https://nixos.org/manual/nix/stable/command-ref/nix-store.html)
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Hash)]
+pub struct NixStoreCmd;
+
+impl NixStoreCmd {
+    pub fn command(&self) -> Command {
+        let mut cmd = Command::new("nix-store");
+        cmd.kill_on_drop(true);
+        cmd
+    }
+}
+
+impl NixStoreCmd {
+    /// Fetch all build and runtime dependencies of given [devour_flake::DrvOut]s
+    ///
+    /// This is done by querying the deriver of each output path from [devour_flake::DrvOut] using [nix_store_query_deriver] and
+    /// then querying all dependencies of each deriver using [nix_store_query_requisites_with_outputs].
+    /// Finally, all dependencies of each deriver are collected and returned as [Vec<StorePath>].
+    pub async fn fetch_all_deps(
+        &self,
+        out_paths: Vec<devour_flake::DrvOut>,
+    ) -> Result<Vec<StorePath>> {
+        let mut all_drvs = Vec::new();
+        for out in out_paths.iter() {
+            let devour_flake::DrvOut(out_path) = out;
+            let drv = self.nix_store_query_deriver(out_path.clone()).await?;
+            all_drvs.push(drv);
+        }
+        let mut all_outs = Vec::new();
+        for drv in all_drvs {
+            let deps = self
+                .nix_store_query_requisites_with_outputs(drv.clone())
+                .await?;
+            all_outs.extend(deps.into_iter());
+        }
+        Ok(all_outs)
+    }
+
+    /// Return the derivation used to build the given build output.
+    async fn nix_store_query_deriver(&self, out_path: PathBuf) -> Result<PathBuf> {
+        let mut cmd = self.command();
+        cmd.args(["--query", "--deriver", &out_path.to_string_lossy().as_ref()]);
```

Note: this command can fail silently with `unknown-deriver`:

```
❯ nix-store --query --deriver /nix/store/50rs0cwcp2878l3hvymwnq6v7cyyw1l3-haskell-template
/nix/store/myd60l138syblrhwckpwxh259cn2s6ms-haskell-template.drv

srid on appreciate.intern nixci on  print-all-deps [!] is 📦 v0.4.0 via 🦀 v1.74.0 via ❄️  impure (nixci-env)
❯ nix-store --query --deriver /nix/store/myd60l138syblrhwckpwxh259cn2s6ms-haskell-template.drv
unknown-deriver

srid on appreciate.intern nixci on  print-all-deps [!] is 📦 v0.4.0 via 🦀 v1.74.0 via ❄️  impure (nixci-env)
❯ echo $?
0
```